ddpui/api/user_org_api.py

```
@@ -137,16 +139,60 @@
+        # Instead, always check by jti and user_id.
+        token_user_id = token.payload.get("user_id")
+        if not request.user or request.user.id != token_user_id:
+            raise HttpError(403, "Token does not belong to the current user")
+        # Blacklist by token object (which uses jti under the hood)
+        token.blacklist()
+        return {"success": 1}
+    except TokenError:
+        # Token is already invalid or expired, do not blacklist
+        return {"success": 1}
+    except Exception as e:
+        raise HttpError(400, f"Logout failed: {str(e)}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)